Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -440,6 +442,26 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
             promise->reject(Exception { ExceptionCode::TypeError, "The cookie name and value must not both be empty."_s });
             return;
         }
+
+        if (cookie.value.startsWithIgnoringASCIICase("__Host-"_s)
+            || cookie.value.startsWithIgnoringASCIICase("__Host-Http-"_s)
+            || cookie.value.startsWithIgnoringASCIICase("__Http-"_s)
+            || cookie.value.startsWithIgnoringASCIICase("__Secure-"_s)) {
+            promise->reject(Exception { ExceptionCode::TypeError, "If the cookie name is empty, the value must not begin with \"__Host-\", \"__Host-Http-\", \"__Http-\", or \"__Secure-\""_s });
+            return;
+        }
+    }
+
+    if (cookie.name.startsWithIgnoringASCIICase("__Host-Http-"_s)
+        || cookie.name.startsWithIgnoringASCIICase("__Http-"_s)) {
+            promise->reject(Exception { ExceptionCode::TypeError, "The cookie name must not begin with \"__Host-Http-\" or \"__Http-\""_s });
+            return;
+    }
+
+    // FIXME: <rdar://85515842> Obtain the encoded length without allocating and encoding.
+    if (cookie.name.utf8().length() + cookie.value.utf8().length() > maximumNameValuePairSize) {
```

Comment:
This adds to the list of places where we need a utf8Length that doesn't allocate or write.